src/mpack/mpack-writer.h

```diff
@@ -765,6 +765,82 @@ MPACK_INLINE void mpack_finish_type(mpack_writer_t* writer, mpack_type_t type) {
     mpack_writer_track_pop(writer, type);
 }
 
+#if defined(__cplusplus)
+#define MPACK_HAS_GENERIC 1
```

Comment:
I tried adding the whole bunch into platform, it didn't feel like a good place for write function and got smacked by the compile it had no clue how the `mpack_writer_t` looks like. So I have cleaned it up a bit: https://github.com/ludocode/mpack/commit/e9a1875ad8afccd6629b0613bc99c57db202b578.